x-pack/plugins/apm/server/lib/transaction_groups/correlations/get_significant_terms_agg.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ESFilter } from '../../../../typings/elasticsearch';
+
+export function getSignificantTermsAgg(
+  fieldNames: string[],
+  backgroundFilters: ESFilter[]
+) {
+  return fieldNames.reduce((acc, fieldName) => {
+    return {
+      ...acc,
+      [fieldName]: {
+        significant_terms: {
+          size: 10,
+          field: fieldName,
+          background_filter: { bool: { filter: backgroundFilters } },
```

Comment:
it would be nice if the type of analysis is also configurable via the API. We could simply stringify/parse JSON and add it to the aggregation. You could also create an io-ts type, something like:

```
const sigTermsOptionsRt = t.union([
	t.type({ percentage: t.strict({}) }),
	t.type({ mutual_information: t.partial({ include_negatives: t.boolean}) }),
	t.type({ script_heuristic: t.type({ script: t.any }) }),
	t.partial({ min_doc_count: t.number })
]);
```

Reply:
I now added the various scoring options as params defaulting to percentage.